test/e2e/framework/dump.go

```diff
@@ -46,58 +47,67 @@ func retrieveContainerLogs(ctx context.Context, podClient corev1client.PodInterf
 
 	_, err = io.Copy(dest, readCloser)
 	if err != nil {
-		return err
+		return fmt.Errorf("can't read logs: %w", err)
 	}
 
 	return nil
 }
 
+func findContainerStatus(containerStatuses []corev1.ContainerStatus, containerName string) *corev1.ContainerStatus {
+	var s *corev1.ContainerStatus
+	for i := range containerStatuses {
+		if containerStatuses[i].Name == containerName {
+			s = &containerStatuses[i]
+			break
+		}
+	}
+	return s
+}
+
 func dumpPodLogs(ctx context.Context, coreClient corev1client.CoreV1Interface, pod *corev1.Pod, resourceDir string, namespace string) error {
 	podDir := filepath.Join(resourceDir, pod.Name)
 	err := os.Mkdir(podDir, 0777)
 	if err != nil {
-		return err
+		return fmt.Errorf("can't make pod dir %q: %w", podDir, err)
 	}
 
 	for _, c := range pod.Spec.Containers {
+		cs := findContainerStatus(pod.Status.ContainerStatuses, c.Name)
+		if cs == nil {
+			return fmt.Errorf("can't find container status for container %q in pod %q", c.Name, naming.ObjRef(pod))
+		}
+
 		logOptions := &corev1.PodLogOptions{
 			Container:  c.Name,
 			Follow:     false,
 			Timestamps: true,
 		}
 
-		// Retrieve current logs.
-		logOptions.Previous = false
-		err = retrieveContainerLogs(ctx, coreClient.Pods(namespace), filepath.Join(podDir, fmt.Sprintf("%s.current", c.Name)), pod.Name, logOptions)
-		if err != nil {
-			return err
-		}
-
-		hasPrevious := false
-		for _, cs := range pod.Status.ContainerStatuses {
-			if cs.Name != c.Name {
-				continue
-			}
-
-			if cs.RestartCount > 0 {
-				hasPrevious = true
+		if cs.State.Running != nil {
```

Comment:
shouldn't we also check the value of running?

Reply:
I don't think we care about the value of the timestamp. Looks like kube checks similarly:
https://github.com/kubernetes/kubernetes/blob/cdf1b17bb8bcbf8a420d7e0a11336d09b0c37730/staging/src/k8s.io/kubectl/pkg/describe/describe.go#L1956-L1958

Reply:
oh i thought it's a `*bool`.